src/pydisagg/ihme/splitter/cat_splitter.py

```diff
@@ -0,0 +1,632 @@
+from typing import Any, List
+
+import numpy as np
+import pandas as pd
+import multiprocessing
+from pandas import DataFrame
+from pydantic import BaseModel
+from typing import Literal
+from joblib import Parallel, delayed
+
+from pydisagg.disaggregate import split_datapoint
+from pydisagg.ihme.schema import Schema
+from pydisagg.models import RateMultiplicativeModel, LogOddsModel
+from pydisagg.ihme.validator import (
+    validate_columns,
+    validate_index,
+    validate_noindexdiff,
+    validate_nonan,
+    validate_positive,
+)
+
+
+class CatDataConfig(Schema):
+    """
+    Configuration for the data DataFrame.
+
+    Parameters
+    ----------
+    index : List[str]
+        List of column names to be used as index in the data DataFrame.
+    target : str
```

Comment:
It is better to give an example here, especially if we are gonna go with the list route.